docs/developers/sc-calls.md

```diff
@@ -0,0 +1,297 @@
+    ///
+    /// Only works with contracts from the same shard.
+    pub fn sync_call_same_context(self) -> <RH::ListReturns as NestedTupleFlatten>::Unpacked
+```
+
+```rust title=adder.rs
+    #[endpoint]
+    fn sync_same_context(&self, destination: ManagedAddress, value: BigUint) {
+        self.tx()
+            .to(destination)
+            .typed(adder_proxy::AdderProxy)
+            .add(value)
+            .sync_call_same_context();
+    }
+```
+
+### Sync call readonly
+
+This type of call performs a synchronous call in `readonly` mode, meaning the destination contract's state cannot be altered by this action. This type of call is performed with `.sync_call_readonly()`. 
+
+```rust
+    /// Executes transaction synchronously, in readonly mode (target contract cannot have its state altered).
+    ///
+    /// Only works with contracts from the same shard.
+    pub fn sync_call_readonly(self) -> <RH::ListReturns as NestedTupleFlatten>::Unpacked
+```
+
+```rust title=adder.rs
+    #[endpoint]
+    fn sync_readonly(&self, destination: ManagedAddress, _value: BigUint) {
+        self.tx()
+            .to(destination)
+            .typed(adder_proxy::AdderProxy)
+            .sum()
+            .sync_call_readonly();
+    }
+```
+
+[comment]: # (mx-context-auto)
+
+### Deploy call
+
+Sync calls can also be used to deploy new contracts. There are two primary methods:
+- Raw deploy (directly deploys a contract with specific bytecode)
+- Deploy from source (deploys a contract using bytecode from an existing contract address)
+
+Syntax-wise, both of these calls are executed using `.sync_call()`, but the transaction setup differs for each type.
+
+For a simple `raw deploy`, initiate a raw deploy transaction as shown below:
+
+```rust title=adder.rs
+    #[endpoint]
+    fn raw_deploy(&self, code: ManagedBuffer) -> ManagedAddress {
+        self.tx()
+            .raw_deploy()
+            .code(code)
+            .code_metadata(CodeMetadata::UPGRADEABLE)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call()
+    }
+```
+
+In the example above, only `.code()` is mandatory for a deploy sync call. We need to either pass the code to the transaction as an argument, or to receive it from a specified location on the blockchain (from source).
+
+In the case of a `deploy from source` transaction, we would use the specific function `.from_source()` instead of `.code()` and pass the source address as a parameter, as such:
+
+```rust title=adder.rs
+    #[endpoint]
+    fn raw_deploy_from_source(&self, source: ManagedAddress) -> ManagedAddress {
+        self.tx()
+            .raw_deploy()
+            .from_source(source)
+            .code_metadata(CodeMetadata::UPGRADEABLE)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call()
+    }
+```
+
+## Async calls
+
+Async calls are used when `cross-shard communication` is required, allowing transactions to execute across multiple shards.
```

Comment:
Point out that they also work in the same shard.

Also talk about them running later and callbacks.